### PR TITLE
[#176591026] Added new stage for deploying client SDK to NPM

### PR DESCRIPTION
**Pull request by gquadrati**
- [x] Added new stage for publishing cliend SDK to NPM

- [ ] Add NPM_CONNECTION variable to Azure DevOps

- [ ] Test SDK deploy to NPM

### CODE REVIEW

File: .devops/deploy-pipelines.yml [hunk 1]
```diff
@@ -53,7 +67,7 @@ resources:
     - repository: pagopaCommons
       type: github
       name: pagopa/azure-pipeline-templates
-      ref: refs/tags/v8
+      ref: refs/tags/v10
```
Comment (balanza) [inline .devops/deploy-pipelines.yml:70]: Optimistic 🤗

Comment (gquadrati) [reply .devops/deploy-pipelines.yml:70]: let me dream

File: .devops/deploy-pipelines.yml [hunk 2]
```diff
@@ -95,6 +109,22 @@ stages:
...
+      vmImage: 'ubuntu-latest'
+    jobs:
+      - ${{ each spec in parameters.OPENAPI_SPEC_PATHS }}:
+        - job: publish_SDK_${{ spec.Key }}   
+          steps:
+          # Template for generating and deploying client SDk to NPM
+          - template: templates/client-sdk-publish/template.yaml@pagopaCommons
+            parameters:
+              openapiSpecPath: '${{ spec.Value }}'
+              artifactName: 'Bundle_${{ spec.Key }}'
```
Comment (balanza) [inline .devops/deploy-pipelines.yml:126]: What do you think the name of each package would be?

**Review by balanza — CHANGES_REQUESTED** [review]

File: .devops/deploy-pipelines.yml (hunk 2 above)
Comment (gquadrati) [reply .devops/deploy-pipelines.yml:126]: > What do you think the name of each package would be?

I think they should be "Bundle api_auth", "Bundle api_backend", ..

File: .devops/deploy-pipelines.yml [hunk 3]
```diff
@@ -206,3 +221,19 @@ stages:
...
```
Comment (balanza) [inline .devops/deploy-pipelines.yml:239]: We should specify the package name for each sdk

**Review by balanza — CHANGES_REQUESTED** [review]

File: .devops/deploy-pipelines.yml (hunk 3 above)
Comment (gquadrati) [reply .devops/deploy-pipelines.yml:239]: > We should specify the package name for each sdk

Can we think about something like _'@pagopa/io-backend-**auth**-sdk'_

Comment (balanza) [reply .devops/deploy-pipelines.yml:239]: We must find a name for each fo the following:
```yaml
api_auth: 'api_auth.yaml'
      api_backend: 'api_backend.yaml'
      api_bonus: 'api_bonus.yaml'
      api_bpd: 'api_bpd.yaml'
      api_myportal: 'api_myportal.yaml'
      api_notifications: 'api_notifications.yaml'
      api_pagopa: 'api_pagopa.yaml'
      api_public: 'api_public.yaml'
      api_session: 'api_session.yaml'
```

Comment (gquadrati) [reply .devops/deploy-pipelines.yml:239]: I konw, I was moving to a list of objects, I'm gonna push a proposal and then we decide

Comment (gquadrati) [reply .devops/deploy-pipelines.yml]: Done in [105db74](https://github.com/pagopa/io-backend/pull/772/commits/105db74a1972d682baf9470f8af578618eaf0cc0)

File: .devops/deploy-pipelines.yml [hunk 4]
```diff
@@ -40,15 +40,33 @@ parameters:
...
-      api_auth: 'api_auth.yaml'
-      api_backend: 'api_backend.yaml'
-      api_bonus: 'api_bonus.yaml'
-      api_bpd: 'api_bpd.yaml'
-      api_myportal: 'api_myportal.yaml'
-      api_notifications: 'api_notifications.yaml'
-      api_pagopa: 'api_pagopa.yaml'
-      api_public: 'api_public.yaml'
-      api_session: 'api_session.yaml'
+      api_auth:
```
Comment (gunzip) [inline .devops/deploy-pipelines.yml:43]: maybe it's better using consistent prefixes like `@pagopa/io-sdk-backend-auth`
(from the wider scope to the tighter)

Comment (gquadrati) [reply .devops/deploy-pipelines.yml:43]: > maybe it's better using consistent prefixes like `@pagopa/io-sdk-backend-auth`
> (from the wider scope to the tighter)

I tried to reuse the patter of the code gen, that adds _"-sdk"_ at the end of the package name

Comment (balanza) [reply .devops/deploy-pipelines.yml:43]: The pattern `io-sdk-<api_name>` makes me think it's somehow related to https://github.com/pagopa/io-sdk, that's why I would avoid it.

> (from the wider scope to the tighter)

That's actually what we thought: 
```
scope of api (application name) --> scope of package ("-sdk")
```
Anyway, so far we aimed to not have to name each package, but being able to infer the name from the application which is serving the api (see https://github.com/pagopa/io-utils/blob/master/src/commands/gen-api-sdk/index.ts#L65). We're still in time to change this for a better pattern, if any proposal exists.

This is not the case of the current application, which exposes several different apis from the application. Hence, we need to name each one.

Comment (balanza) [reply .devops/deploy-pipelines.yml:43]: Any thoughts? Can we proceed with those names?

**Review by balanza — APPROVED** [review]

**Pull request merged**